sys/include/net/nanocoap.h

```diff
@@ -847,18 +847,18 @@ static inline ssize_t coap_get_uri_path(coap_pkt_t *pkt, uint8_t *target)
  * This function decodes the pkt's URI_QUERY option into a "&"-separated and
  * '\0'-terminated string.
  *
- * Caller must ensure @p target can hold at least CONFIG_NANOCOAP_URI_MAX bytes!
- *
  * @param[in]   pkt     pkt to work on
```

Comment:
```suggestion
 * @note This function is mainly intended to get easy debug output.
 * It is needlessly complex and error prone for parsing parameters out of the query,
 * which is better done by @ref coap_opt_get_opaque with increasing values of @p opt_num,
 * until that returns `-ENOENT`.

 * @param[in]   pkt     pkt to work on
```

Part of the footgun (IMO the bigger footgun even) is that this function drives people toward assembling and later parsing this string, which inevitably leads to trouble once the first `&` appears in a query parameter (and if that were fixed, the string would go through percent encoding and then back again).

Reply:
Scratch that -- we'll still need such a note, but https://github.com/RIOT-OS/RIOT/pull/20197 adds something that would be pointed to instead.